WonderMake/System/System.h

```diff
@@ -1,19 +1,69 @@
 #pragma once
-#include "Message/MessageSubscriber.h"
+
+#include "Policies/Policy.h"
+
+#include "System/SystemContainer.h"
+
+#include "Utilities/RestrictTypes.h"
+
+#include <optional>
 
 class SystemContainer;
 
-class System 
+class SystemBase
 	: public NonCopyable
 {
 public:
-	virtual ~System() = default;
+	virtual ~SystemBase() = default;
 
-	virtual void Tick() noexcept {}
+	virtual void Tick() {}
 
 protected:
 	void EnableTick();
 
-	System() = default;
+	constexpr SystemBase() noexcept = default;
+};
+
+template<typename TPolicySet = Policy::Set<>>
+class System
+	: public SystemBase
+{
+public:
+	using Super = System<TPolicySet>;
+	using Dependencies = typename TPolicySet::template DependenciesRef;
+
+	inline constexpr System(Dependencies&& aDependencies) noexcept;
+
+protected:
+	constexpr System() noexcept = default;
+	
+	template<
+		typename TDependency,
+		typename = std::enable_if_t<
+			TPolicySet::template HasPolicy_v<TDependency, Policy::EPermission::Write>
+			|| TPolicySet::template HasPolicy_v<TDependency, Policy::EPermission::Unrestricted>>>
```

Comment:
requires? Or did it not work properly here?

Reply:
I do not remember, will have to check it out